### PR TITLE
Spiral\Scheduler\Job\Job::filtersPass(): Argument #1 ($container) must be of type Spiral\Core\Container #21

### PR DESCRIPTION
**Pull request by butschster**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "butschster/cron-expression-generator": "^1.10",
     "nesbot/carbon": "^2.52 || ^3.0",
     "psr/event-dispatcher": "^1",
+    "spiral/core": "^3.0",
```
Comment (roxblnfk) [inline composer.json:23]: ^3.14 for all deps

File: src/JobsLocator.php [hunk 2]
```diff
@@ -44,10 +45,10 @@ private function createJob(\ReflectionClass $class, ScheduleAttribute $schedule)
                 $schedule->expression
             ),
             $schedule->description,
-            static function (Container $container) use ($className, $parameters) {
-                $object = $container->make($className);
+            static function (FactoryInterface $factory, InvokerInterface $invoker) use ($className, $parameters) {
+                $object = $factory->make($className);
 
-                $container->invoke([
+                $invoker->invoke([
```
Comment (roxblnfk) [inline src/JobsLocator.php:51]: I think you can avoid using factory there

```php
$invoker->invoke([$className, $method])
```

It should resolve $className from container.

Or you don't need caching there?

**Review by roxblnfk — APPROVED** [review]
Feel free to merge if it's working

**Review by daniellienert — APPROVED** [review]
Not experienced enough with the framework internals to judge the code changes - but the change fixes the scheduler issue in our projects. 

Would be great to get it merged  & released

**Comment by roxblnfk** [comment]
@butschster can we merge?

**Pull request merged**